Modules/Core/Common/include/itkImageBase.h

```diff
@@ -768,6 +768,19 @@ class ITK_TEMPLATE_EXPORT ImageBase : public DataObject
   bool
   VerifyRequestedRegion() override;
 
+  /** Check whether this image and the other image occupy the same physical space.
+   * Compares the origin, spacing, and direction for equality within provided tolerances. */
+  bool
+  IsCongruentImageGeometry(const ImageBase * otherImage, double coordinateTolerance, double directionTolerance) const;
+
+  /** Check whether this image and the other image have the same grid in physical space.
+   * Compares largest possible regions for equality, and the origin, spacing,
+   * and direction cosines for equality within provided tolerances. */
+  bool
+  IsSameImageGeometryAs(const ImageBase * otherImage,
+                        double            coordinateTolerance = 1e-6,
+                        double            directionTolerance = 1e-6) const;
```

Comment:
Perhaps we should just have no defaults here, and have a documentation link to the ImageToImage defaults?

Reply:
That will be an extra step for common use cases. I think we should keep the defaults, but add a link to the `ImageToImage` defaults, when someone needs to learn about the tolerances.

Reply:
Perhaps we could use the same global constant is [both](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/src/itkImageToImageFilterCommon.cxx#L26) places then?

Maybe add a common constant value in Macro.h?

Reply:
That is a good idea. But I am already tired of this small PR. Can we do that in a different PR? Otherwise it would be good if someone took over this PR.